File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        list1 = list(word1)
        list2 = list(word2)
        answer = ""
        while list1 or list2:
            if list1:
                answer += list1.pop(0)
            if list2:
                answer += list2.pop(0)

        return answer
        
        
        

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder s = new StringBuilder();
        
        while(word1.length() > 0 || word2.length() > 0)
        {
            if(word1.length() > 0)
            {
                s.append(word1.charAt(0));
                word1 = word1.substring(1);
            }

            if(word2.length() > 0 )
            {
                s.append(word2.charAt(0));
                word2 = word2.substring(1);
            }
        }
        return s.toString();
    }
}